contracts/Remittance.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity >=0.4.21 <0.6.0;
+
+import "./Pausable.sol";
+
+contract Remittance is Pausable {
+  address public owner;
+  address public carol;
+  bytes32 private withdrawHash;
+
+  event LogDeposited(address indexed sender, uint256 amount, bytes32 hash, address indexed carol);
+  event LogWithdrawn(address indexed sender, uint256 amount);
+
+  constructor(bool paused) Pausable(paused) public {
+  }
+
+  function calcuateHash(address _carol, bytes32 _bobPassword, bytes32 _carolPassword) public view returns(bytes32) {
+     return keccak256(abi.encodePacked(this, _carol, _bobPassword, _carolPassword));
+  }
+
+  function deposit(bytes32 _hash, address _carol) external payable whenRunning whenAlive {
```

Comment:
Can you see how easy it would be for me to steal whoever used this contract?

Reply:
Do you mean adding `onlyOwner`?

Reply:
If you add onlyOwner, you are only allowing the owner to fool other people. That's better than outright stealing, but not good enough.

Reply:
I think it's the same issue in this comment https://github.com/JakeLin/ethereum-remittance/pull/2#discussion_r316921467

I am thinking to create an array or mapping to recode each TX to keep the amount, then no previous TX can be overwritten.

Reply:
Yes, something has to be remembered.